compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -183,6 +183,14 @@ public boolean checkForUniqueSymbol(DiagnosticPos pos, SymbolEnv env, BSymbol sy
             foundSym = lookupSymbolForDecl(env, symbol.name, SymTag.MAIN);
         }
 
+        if (foundSym == symTable.notFoundSymbol && symbol.tag == SymTag.FUNCTION) {
+            int dotPosition = symbol.name.value.indexOf('.');
```

Comment:
This logic may not work when we have a dot with a quoted identifier. Please create an issue.

Reply:
Issue created https://github.com/ballerina-platform/ballerina-lang/issues/26452